### DISCUSSION TITLE
Can this API Silence/Unsilence motion events?

### DISCUSSION THREAD

**Discussion by BillyCroan** [dgreif/ring#1346] [Q&A]
I'm trying to make my ring doorbell suck less.

I live in a house with multiple adults and multiple dogs.  None of us need our phones to beep any time one of us walks one of the dogs.

So I put a hall-effect sensor in the door frame on an RPI.  And I was about to us that to DROP/ACCEPT the ring's traffic in the home firewall.  But I came across this API and I thought maybe it would be cleaner to use an API to adjust notification settings in realtime than to block traffic.

My goal is that if the door has been opened, motion alerts stop.  It should still serve the purpose of alerting me to visitors.  But if someone inside goes outside first, then no alerts.

**Comment by dgreif** [comment]
Yes and no. The challenge with phone alerts is that those settings are stored _per device_. Another device (in this case the ring-client-api) cannot directly control the notifications on another device (your phone).

That said, this api _can_ control the Location Mode settings. You might be able to get creative with the "Home"/"Away"/"Disarmed" settings so that they send the alerts in Home/Away mode, but not in Disarmed. Then you could use this API to temporarily enter Disarmed mode when the door opens. It's not a perfect solution, but might work better than your packet blocking at the network layer 😄

**Comment by tsightler** [comment]
Another alternative, if you don't want to use Modes, you can completely enable/disable motion detection while the door is open.  In [ring-mqtt](https://github.com/tsightler/ring-mqtt) I implemented this as a per-camera switch visible in Home Assistant to toggle motion detection on/off, same as the switch in the Ring app.  This is probably a cleaner solution than blocking network traffic as it will also appear clearly in the Ring app that motion detection is enabled/disabled.  You can see the code to do this in [setMotionDetectionState()](https://github.com/tsightler/ring-mqtt/blob/cc4746e270b79a799097a618c79b1eb5f4fa7703/devices/camera.js#L1165) function in ring-mqtt.

**Reply by drheck** [reply]
I am also looking for this feature, because i have 3 ring cameras and i want in some cases disable motion detection on one of these like tsightler describes!
@dgreif: Will you add this feature in the next version?

**Reply by tsightler** [reply]
This feature is not currently planned for homebridge-ring.  I think we'd be open to reviewing a PR if someone wants to code it up and submit it.  Best alternative options are to use the modes method mentioned above, or to use another project that implements this.

**Reply by drheck** [reply]
Okay. Thank you. I understand that.
I spent many hours today trying to add a switch to toggle the mode for each camera.
Unfortunately I have not found the right place (ring-platform.js or in the camera.js) for it. There has to be a corresponding button for each camera.
Can someone please give me a few tips?
Can i use the switch from switch.js or is it better to build my own?

I've already written 2 plugins for myself privately, but I've put them together using examples and so on. In these, the switches and sensors were created statically and not dynamically. 

Many thanks for your support.

**Reply by tsightler** [reply]
camera.ts is where it should go, just a new `registerCharacteristic` function with the proper values and name, and a getValue/setValue function which does the right thing.  You can use ring-mqtt as a reference since it already has functions for this, specifically (command is either "on" or "off"):

```
await this.device.setDeviceSettings({
    "motion_settings": {
        "motion_detection_enabled": Boolean(command === 'on')
    }
})
```

You could probably just copy the siren switch, change the name and setValue/getValue functions.

**Reply by drheck** [reply]
Thanks. I will try it. :-)

**Reply by drheck** [reply]
Hi, thank you. I have now a button and can set the value.
But i can not get the state!?
The res variable has following value: [27.11.2023, 20:45:32] [Ring] DPH: GETMotDetValue: [object Promise]
Any idea?
Thank you.
```
this.registerCharacteristic({
characteristicType: Characteristic.On,
serviceType: Service.Switch,
serviceSubType: 'MotDet',
name: device.name + ' MotDet',
getValue: () => {
	const res = device.getDeviceSettings();
	const sres = String(res);
	(0, util_1.logInfo)(`DPH: GETMotDetValue: ${sres}`);
	return false;
	//return Boolean(data.siren_status && data.siren_status.seconds_remaining);
},
setValue: (value) => {
	(0, util_1.logInfo)(`DPH: SETMotDetValue: ${value}`);
	device.setDeviceSettings({
		"motion_settings": {
			"motion_detection_enabled": Boolean(value)
		}
	});
},
	requestUpdate: () => device.requestUpdate(),
});

```

**Reply by tsightler** [reply]
I'd guess it would just be something like:

`return data.settings.motion_detection_enabled`

**Reply by drheck** [reply]
Hm. I have tried a lot of things, but no success.
It seems to be a problem with an not asynchronous call...
The variable data is not available...
Perhaps the function: const res = device.getDeviceSettings() is not working?!
Any further idea?
Thank you very much :-)

**Reply by tsightler** [reply]
You don't need that function, I don't know why it is in there.

**Reply by drheck** [reply]
Hm. But how can i read the actual state from the camera?

**Reply by tsightler** [reply]
Look at the examples of other registered characteristics in the code.  The data object always reflects the current state, you don't need to get it, the `data.settings.motion_detection_enabled` property already exist and has the state, you just need to return it

**Reply by drheck** [reply]
Okay. I got it... Thank you. Good night...

**Reply by drheck** [reply]
One further question: Is there a setting for disabeling the motion notifications only (like in the ring app)?
And where can i get the ring api documentation?
